query-engine/prisma-models/src/field/relation.rs

```diff
@@ -47,21 +47,11 @@ impl RelationField {
     }
 
     pub fn scalar_fields(&self) -> Vec<ScalarFieldRef> {
-        let model = self.model();
-        let fields = model.fields();
         self.walker()
             .fields()
             .into_iter()
             .flatten()
-            .map(|f| {
-                fields.find_from_scalar(f.name()).unwrap_or_else(|_| {
-                    panic!(
-                        "Expected '{}' to be a scalar field on model '{}', found none.",
-                        f.name(),
-                        model.name()
-                    )
-                })
-            })
+            .map(|f| self.dm.clone().zip(ScalarFieldId::InModel(f.id)))
```

Comment:
The changes in this file are not strictly necessary, but it's less code and they made a very noticeable difference in the benchmark numbers (~6% IIRC).